lib/src/exception.dart

```diff
@@ -81,10 +81,16 @@ body::before {
 class SassRuntimeException extends SassException {
   final Trace trace;
 
-  SassRuntimeException(String message, FileSpan span, this.trace)
+  SassRuntimeExceptionType type;
+
+  SassRuntimeException(String message, FileSpan span, this.trace, [this.type])
       : super(message, span);
 }
 
+enum SassRuntimeExceptionType {
```

Comment:
Unintuitively, it's generally *not* good practice to use Dart's native enum type. It's quite inflexible once you have to do anything beyond the absolute basics, so rather than rewriting enums entirely once that comes up we generally manually write pseudo-enums from the beginning. See [`ListSeparator`](https://github.com/sass/dart-sass/blob/master/lib/src/value/list.dart#L59-L85) for an example of the manual enum pattern.

I think we may be able to get away with not using an enum here, though. It's only being used to detect errors thrown by `@error`, but I think you can determine that by checking `error.span.text.startsWith("@error")`.